### PR TITLE
rm sponsored and verified categories

### PR DESCRIPTION
**Pull request by eviljeff**
Fixes mozilla/addons#15038

Removes references to Verified and Sponsored promoted groups.  They've never been used in production but we (confusingly) offer them as filtering choices, and supported badging for them, etc.

### Testing
tl;dr make sure nothing is broken by removing obsolete code
- hero area (homepage) still correctly labels add-ons are By Firefox or Recommended
- Search/category pages don't offer verified or sponsored as choices in badging select
- badging filter works as expected
- add-ons that are in recommended or By Firefox promoted groups get a badge on their listing page

### CODE REVIEW

File: src/amo/components/HeroRecommendation/index.js [hunk 1]
```diff
@@ -233,7 +233,7 @@ export class HeroRecommendationBase extends React.Component<InternalProps> {
           // L10n: If uppercase does not work in your locale, change it to lowercase. This is used as a secondary heading.
           titleText = i18n.gettext('BY FIREFOX');
         } else {
-          titleText = i18n.gettext('SPONSORED');
+          titleText = i18n.gettext('PROMOTED');
```
Comment (eviljeff) [inline src/amo/components/HeroRecommendation/index.js:236]: This should never happen (addons-server shouldn't allow it), but it's a safe error state - better than defaulting to one of the other groups like Recommended or By Firefox - and less complicated than trying to handle it at this point in the render.

Comment (KevinMind) [reply src/amo/components/HeroRecommendation/index.js:236]: If it should not happen, let's add a sentry log when it does happen so at least we know something wrong is happening.

**Comment by KevinMind** [comment]
Can you add testing instructions in the PR or the issue? Do we have any checks in addons-frontend to verify dead code is removed?

File: src/amo/components/HeroRecommendation/index.js (hunk 1 above)
Comment (eviljeff) [reply src/amo/components/HeroRecommendation/index.js:236]: we don't have sentry integration for frontend (it was way too noisy)

**Comment by eviljeff** [comment]
> Can you add testing instructions in the PR or the issue?
added

> Do we have any checks in addons-frontend to verify dead code is removed?

Not 100% what you mean.  We have code-cov for test coverage, but that's all, afaik

**Comment by KevinMind** [comment]
Can you define 

> badging filter works as expected

How am I expecting it to work?

I can verify the filter select for badging does not contain the removed categories

Screenshot 2024-09-24 at 17 42 00

**Comment by eviljeff** [comment]
> Can you define
> 
> > badging filter works as expected
> 
> How am I expecting it to work?
> 
> I can verify the filter select for badging does not contain the removed categories

Looking at prod it seems we didn't offer them as choices anyway. Check the other options (recommended, etc) work and return the expected values (if your local instance is connected to amo:stage it would return the same results as on https://addons.allizom.org/)

File: src/amo/components/HeroRecommendation/index.js (hunk 1 above)
Comment (KevinMind) [reply src/amo/components/HeroRecommendation/index.js:236]: Too bad.

**Pull request merged**